pages/edit/[id].js

```diff
@@ -1,28 +1,34 @@
 import TransactionEntryForm from "@/components/TransactionEntryForm";
 import { useRouter } from "next/router";
+import { useSession } from "next-auth/react";
 
 export default function EditPage({ handleEditTransaction, transactions }) {
-	const router = useRouter();
-	const { id } = router.query;
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: session, status } = useSession();
 
-	if (!id) {
-		return null;
-	}
+  if (!id) {
+    return null;
+  }
+    if (status !== "authenticated") {
+      return <h2>Access denied!</h2>;
+    }
```

Comment:
didn't you create a ProtectedPage component which does exactly this? Why don't you use it here?